sdk/storage/azure-storage-file-share/tests/test_directory.py

```diff
@@ -427,6 +427,42 @@ def test_list_subdirectories_and_files(self, storage_account_name, storage_accou
         self.assertEqual(list_dir[5]['name'], 'file3')
         self.assertEqual(list_dir[5]['is_directory'], False)
 
+    @FileSharePreparer()
+    def test_list_subdirectories_and_files_encoded(self, storage_account_name, storage_account_key):
```

Comment:
I think you just migrated this file to test proxy? If so, you'll need to merge main in here and update these tests :).

Also, okay to do in another PR after this merges.

Reply:
Will do! I somehow have managed to merge main into the feature branch and then the feature branch into this branch without alerting all of Microsoft 😄 so will do it all in this PR!